code/Fluent.php

```diff
@@ -669,4 +669,40 @@ public static function get_template_global_variables()
             )
         );
     }
+
+    /**
+     * Given a field on an object and optionally a locale, compare its locale value against the default locale value to
+     * determine if the value is changed at the given locale.
+     *
+     * @param  DataObject  $object
+     * @param  FormField   $field
+     * @param  string|null $locale Optional: if not provided, will be gathered from the request
+     * @return boolean
+     */
+    public static function isFieldModified(DataObject $object, FormField $field, $locale = null)
+    {
+        if (is_null($locale)) {
+            $locale = self::current_locale();
+        }
+
+        if ($locale === $defaultLocale = self::default_locale()) {
+            // It's the default locale, so it's never "modified" from the default locale value
+            return false;
+        }
+
+        $defaultField = self::db_field_for_locale($field->getName(), $defaultLocale);
+        $localeField  = self::db_field_for_locale($field->getName(), $locale);
+
+        $defaultValue = $object->$defaultField;
+        $localeValue  = $object->$localeField;
+
+        if ((!is_null($defaultValue) && is_null($localeValue))
```

Comment:
Sorry, removed my previous second reply... also need to treat `0` as empty values, even if they are strings. :P